whisper.py

```diff
@@ -0,0 +1,100 @@
+from models import (
+    WhisperASRInput,
+    WhisperASROutput,
+    OutputType,
+)
+import logging
+import time
+import json
+from dane.config import cfg
+from dane.provenance import Provenance
+from io_util import (
+    get_base_output_dir,
+    get_output_file_path,
+)
+from faster_whisper import WhisperModel
+
+
+def run_whisper(
+    input: WhisperASRInput,
+) -> WhisperASROutput:
+    logger = logging.getLogger(__name__)
+    logger.info("Starting model application")
+    start = time.time() * 1000  # convert to ms
+    destination = get_output_file_path(input.source_id, OutputType.TRANSCRIPT)
+
+    # float16 only works on GPU, float32 or int8 are recommended for CPU
+    model = WhisperModel(
+        cfg.WHISPER_ASR_SETTINGS.MODEL_VERSION,
+        device=cfg.WHISPER_ASR_SETTINGS.DEVICE,
+        compute_type=(
+            "float16" if cfg.WHISPER_ASR_SETTINGS.DEVICE == "cuda" else "float32"
+        ),
+    )
+
+    segments, _ = model.transcribe(
+        input.input_file_path,
+        vad_filter=cfg.WHISPER_ASR_SETTINGS.VAD,
+        beam_size=5,
+        best_of=5,
+        temperature=(0.0, 0.2, 0.4, 0.6, 0.8, 1.0),
```

Comment:
Would it make sense to make these hard-coded values configurable? If not, a bit of explanation would be good to add.

Reply:
Yeah, making them configurable makes sense. As for where people can read more about this, I am not sure exactly. I think the official page for OpenAI's Whisper is a good starting point.